### PR TITLE
James 1800 Conversion from JSON filter to POJO

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

**Comment by aduprat** [comment]
:+1:

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/json/FilterDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    private Class<? extends Filter> detectClass(Iterator<Map.Entry<String, JsonNode>> elements) {
+        Optional<Class<? extends Filter>> possibleFilterOperator = Iterators.toStream(elements)
+                .map(Map.Entry::getKey)
+                .filter(name -> name.equals("operator"))
+                .findFirst()
+                .map(x -> FilterOperator.class);
+
+        return possibleFilterOperator.orElse(FilterCondition.class);
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/json/FilterDeserializer.java]: I think we usually use maybe for that : `maybeFilterOperator`

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/FilterTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                .build();
+        Filter actual = parser.readValue(json, Filter.class);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void singleFilterConditionShouldBeDeserialized() throws Exception {
+        String json = "{\"inMailboxes\": [\"12\",\"34\"]}";
+        Filter expected = FilterCondition.builder()
+                .inMailboxes(Optional.of(ImmutableList.of("12","34")))
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/FilterTest.java]: I suggest to propose a more user-friendly API in FilterConditionBuilder

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/json/FilterDeserializer.java (hunk 1 above)
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/json/FilterDeserializer.java]: done

**Comment by mbaechler** [comment]
replaced by https://github.com/linagora/james-project/pull/335

**Pull request closed without merging**